pkg/testing/utils/utils.go

```diff
@@ -176,3 +177,22 @@ func GenerateTestNamespaceName(t *testing.T, namespace string) string {
 	t.Logf("Seed used: %d", seed)
 	return fmt.Sprintf("%s-%d", namespace, r.Int())
 }
+
+const (
+	NormalizedInt = 1
+	NormalizedStr = "foo"
+)
+
+func NormalizeInt[T constraints.Integer](f *T) {
+	var zero T
+
+	if *f != zero {
+		*f = T(NormalizedInt)
+	}
+}
+
+func NormalizeString(f *string) {
+	if *f != "" {
+		*f = NormalizedStr
+	}
+}
```

Comment:
It's what it does, if the field is not present in the json, it'll be unmarshaled as `0`/`""` and the test will fail because it expects `1`/`"foo"`:
https://github.com/inspektor-gadget/inspektor-gadget/blob/300055ca18c7e18bcf78690959e758cdaaee2f89/gadgets/trace_mount/test/trace_mount_test.go#L107-L113